pkg/nsx/services/ipaddressallocation/ipaddressallocation.go

```diff
@@ -127,6 +130,73 @@ func (service *IPAddressAllocationService) CreateOrUpdateIPAddressAllocation(obj
 	return true, nil
 }
 
+// CreateIPAddressAllocationIfNeeded is only for the restore of external address binding.
+func (service *IPAddressAllocationService) CreateIPAddressAllocationIfNeeded(addressBinding *v1alpha1.AddressBinding, subnetPort *v1alpha1.SubnetPort, restoreMode bool) error {
+	if !restoreMode {
+		// Only create the NSX IPAddressAllocation in restore stage
+		return nil
+	}
+	if len(addressBinding.Status.IPAddress) == 0 {
+		return nil
+	}
+	existingIPAddressAllocation, err := service.GetIPAddressAllocationByOwner(addressBinding)
+	if err != nil {
+		log.Error(err, "Failed to get IPAddressAllocation for AddressBinding", "AddressBinding", addressBinding)
+		return err
+	}
+	if existingIPAddressAllocation != nil {
+		log.V(1).Info("The IPAddressAllocation has been created, skipping", "AddressBinding", addressBinding)
+		return nil
+	}
+	nsxIPAddressAllocation, err := service.BuildIPAddressAllocation(addressBinding, subnetPort, restoreMode)
+	if err != nil {
+		return err
+	}
+	if nsxIPAddressAllocation == nil {
+		return fmt.Errorf("failed to build NSX IPAddressAllocation for AddressBinding %s", addressBinding.UID)
+	}
+	err = service.Apply(nsxIPAddressAllocation)
+	if err != nil {
+		log.Error(err, "Failed to create NSX IPAddressAllocation for AddressBinding", "AddressBinding", addressBinding)
+		return err
+	}
+	log.Info("Successfully created NSX IPAddressAllocation", "nsxIPAddressAllocation", nsxIPAddressAllocation)
+	return nil
+}
+
+// DeleteIPAddressAllocationIfExists is to delete the NSX IPAddressAllocation for the AddressBinding generated during restore mode.
+// It's only invoked when the SubnetPort CR is deleted.
```

Comment:
Update the comment here as this function also is called here for AddressBinding CR deletion?
https://github.com/vmware-tanzu/nsx-operator/blob/b327e8ce18780a5fb6111f37bb6db7234dcbb438/pkg/controllers/subnetport/subnetport_controller.go#L150-L154

Reply:
Sure, thanks.